packages/mobile/src/verify/reducer.ts

```diff
@@ -0,0 +1,340 @@
+    })
+    .addCase(startKomenciSession, (state) => {
+      return {
+        ...state,
+        currentState: startingKomenciSession(),
+      }
+    })
+    .addCase(fetchPhoneNumberDetails, (state) => {
+      return {
+        ...state,
+        currentState: fetchingPhoneNumberDetails(),
+      }
+    })
+    .addCase(setUseKomenci, (state, action) => {
+      return {
+        ...state,
+        shouldUseKomenci: action.payload,
+        komenci: initialState.komenci,
+      }
+    })
+    .addCase(setPhoneHash, (state, action) => {
+      return {
+        ...state,
+        phoneHash: action.payload,
+      }
+    })
+    .addCase(fetchMtw, (state) => {
+      return {
+        ...state,
+        currentState: fetchingMtw(),
+      }
+    })
+    .addCase(fetchOnChainData, (state) => {
+      return {
+        ...state,
+        currentState: fetchingOnChainData(),
+      }
+    })
+    .addCase(setVerificationStatus, (state, action) => {
+      return {
+        ...state,
+        status: {
+          ...state.status,
+          ...action.payload,
+        },
+      }
+    })
+    .addCase(setKomenciContext, (state, action) => {
+      return {
+        ...state,
+        komenci: {
+          ...state.komenci,
+          ...action.payload,
+        },
+      }
+    })
+    .addCase(setActionableAttestation, (state, action) => {
+      return {
+        ...state,
+        actionableAttestations: action.payload,
+      }
+    })
+    .addCase(setOverrideWithoutVerification, (state, action) => {
+      return {
+        ...state,
+        TEMPORAR_override_withoutVerification: action.payload,
+      }
+    })
+    .addCase(checkIfKomenciAvailable, (state) => {
+      return {
+        ...state,
+        komenciAvailable: undefined,
+      }
+    })
+    .addCase(setKomenciAvailable, (state, action) => {
+      return {
+        ...state,
+        komenciAvailable: action.payload,
+      }
+    })
+    .addCase(reset, (state, action) => {
+      return {
+        ...initialState,
+        e164Number: state.e164Number,
+        status: {
+          ...initialState.status,
+          komenci: action.payload.komenci,
+        },
+        komenciAvailable: action.payload.komenci,
+      }
+    })
+})
+
+const isBalanceSufficientForAttestations = (state: RootState, attestationsRemaining: number) => {
+  const userBalance = stableTokenBalanceSelector(state) || 0
+  return new BigNumber(userBalance).isGreaterThan(
+    attestationsRemaining * ESTIMATED_COST_PER_ATTESTATION
+  )
+}
+
+export const currentStateSelector = (state: RootState) => state.verify.currentState
+export const e164NumberSelector = (state: RootState) => state.verify.e164Number
+export const phoneHashSelector = (state: RootState) => state.verify.phoneHash
+export const komenciContextSelector = (state: RootState) => state.verify.komenci
+export const shouldUseKomenciSelector = (state: RootState) => {
+  if (state.verify.komenciAvailable === undefined) {
+    return undefined
```

Comment:
WDYT about returning an enum instead of true|false|undefined?

Reply:
That way it's more explicit about what this case means

Reply:
I will take a look on how TS serialize enums and update it if that would be convenient!